sql/moz-fx-data-shared-prod/analysis/unified_metrics_v1/query.sql

```diff
@@ -0,0 +1,258 @@
+    days_since_created_profile,
+    NULL AS days_since_seen_session_start,
+    NULL AS days_since_seen_session_end,
+    days_seen_bits,
+    days_created_profile_bits,
+    first_seen_date AS first_run_date,
+    durations,
+    NULL AS days_seen_session_start_bits,
+    NULL AS days_seen_session_end_bits,
+    os AS normalized_os,
+    osversion AS normalized_os_version,
+    NULL AS android_sdk_version,
+    locale,
+    city,
+    country,
+    app_build_id AS app_build,
+    NULL AS app_channel,
+    metadata_app_version AS app_display_version,
+    NULL AS architecture,
+    NULL AS device_manufacturer,
+    device AS device_model,
+    NULL AS telemetry_sdk_build,
+    first_seen_date,
+    submission_date = first_seen_date AS is_new_profile,
+    NULL AS n_metrics_ping,
+    NULL AS days_sent_metrics_ping_bits,
+    NULL AS uri_count,
+    default_browser AS is_default_browser,
+    distribution_id AS distribution_id,
+    'Focus Android' AS normalized_app_name,
+    NULL AS isp
+  FROM
+    telemetry.core_clients_last_seen
+  WHERE
+    submission_date = @submission_date
+    AND app_name = 'Focus'
+    AND os = 'Android'
+),
+unioned AS (
+  SELECT
+    * REPLACE (
+      -- Per bug 1757216 we need to exclude BrowserStack clients from KPIs,
+      -- so we mark them with a separate app name here. We expect BrowserStack
+      -- clients only on release channel of Fenix, so the only variant this is
+      -- expected to produce is 'Fenix BrowserStack'
+      IF(
+        isp = 'BrowserStack',
+        CONCAT(normalized_app_name, ' BrowserStack'),
+        normalized_app_name
+      ) AS normalized_app_name
+    )
+  FROM
+    unioned_source
+),
+search_clients AS (
+  SELECT
+    *
+  FROM
+    search_derived.mobile_search_clients_daily_v1
+  WHERE
+    submission_date = DATE_ADD(@submission_date, INTERVAL 1 DAY)
+),
+search_metrics AS (
+  SELECT
+    unioned.client_id,
+    unioned.submission_date,
+        -- the table is more than one row per client (one row per engine, looks like), so we have to aggregate.
+    SUM(ad_click) AS ad_click,
+    SUM(organic) AS organic_search_count,
+    SUM(search_count) AS search_count,
+    SUM(search_with_ads) AS search_with_ads,
+  FROM
+    unioned
+  LEFT JOIN
+    search_clients m
+  ON
+    unioned.client_id = m.client_id
+    AND DATE_ADD(unioned.submission_date, INTERVAL 1 DAY) = m.submission_date
+  GROUP BY
+    1,
+    2
+),
+mobile_with_searches AS (
+  SELECT
+    unioned.client_id,
+    unioned.sample_id,
+    CASE
+    WHEN
+      BIT_COUNT(days_seen_bits)
+      BETWEEN 1
+      AND 6
+    THEN
+      'infrequent_user'
+    WHEN
+      BIT_COUNT(days_seen_bits)
+      BETWEEN 7
+      AND 13
+    THEN
+      'casual_user'
+    WHEN
+      BIT_COUNT(days_seen_bits)
+      BETWEEN 14
+      AND 20
+    THEN
+      'regular_user'
+    WHEN
+      BIT_COUNT(days_seen_bits) >= 21
+    THEN
+      'core_user'
+    ELSE
+      'other'
+    END
+    AS activity_segment,
```

Comment:
Is this activity data (days_seen_bits) only available in this table? I just wonder if it would be worth extracting this logic into a UDF so that the same logic can be applied consistently when calculating `activity_segment`.

Reply:
Is available in the client_last_seen tables, I agree is better in a UDF.